app/presenters/outcome_presenter.rb

```diff
@@ -1,7 +1,25 @@
 require 'erubis'
 
 class OutcomePresenter < NodePresenter
-  class OutcomeTemplateMissing < StandardError; end
+  class ViewContext
+    def initialize(state)
+      @state = state
+      state_methods = state.methods - SmartAnswer::State.new(nil).methods
+      @state_reader_methods = state_methods.reject { |m| m =~ /=$/ }
+    end
+
+    def method_missing(method, *args, &block)
+      if @state_reader_methods.include?(method)
+        @state.send(method, *args, &block)
+      else
+        super
+      end
+    end
+
+    def get_binding
```

Comment:
I suspect @chrisroos has taken this from the [example in the Ruby docs](http://ruby-doc.org/core-2.2.2/Binding.html). I think it might be confusing or even cause a problem to use `#binding`.